baseplate/sidecars/trace_publisher.py

```diff
@@ -87,29 +110,34 @@ def publish(self, payload: SerializedBatch) -> None:
         for _ in RetryPolicy.new(attempts=self.retry_limit):
             try:
                 with self.metrics.timer("post"):
-                    response = self.session.post(
-                        self.endpoint,
-                        data=payload.serialized,
-                        headers=headers,
-                        timeout=self.post_timeout,
-                        stream=False,
-                    )
+                    with publishLatency.time():
+                        response = self.session.post(
+                            self.endpoint,
+                            data=payload.serialized,
+                            headers=headers,
+                            timeout=self.post_timeout,
+                            stream=False,
+                        )
                 response.raise_for_status()
             except requests.HTTPError as exc:
+                publishErrors.labels(error_type="http").inc()
```

Comment:
This seems quite strange to me in comparison with the other metrics we use around requests. Why should we track all errors in one bucket, then a subset in another bucket? Why not have one metric that identifies the type of error (then you can sum by error type) like how the requests lib tracks response code?